src/pushsource/_impl/model/productid.py

```diff
@@ -1,14 +1,114 @@
+from collections import defaultdict
+
+from cryptography import x509
+from frozenlist2 import frozenlist
+from pyasn1.codec.der import decoder
+
 from .base import PushItem
+from .conv import convert_maybe, sloppylist
 from .. import compat_attr as attr
 
 
+# Red Hat OID namespace is "1.3.6.1.4.1.2312.9",
+# the trailing ".1" designates a Product Certificate.
+OID_NAMESPACE = "1.3.6.1.4.1.2312.9.1."
+
+
+@attr.s()
+class ProductId(object):
+    """A ProductID represents a group of metadata pertaining to a single product
+    contained in a ProductID certificate."""
+
+    id = attr.ib(type=int)
+    """Product Engineering ID (EngID), e.g. 72
+
+    :type: int
+    """
+
+    name = attr.ib(type=str, default=None)
+    """Human readable product name, e.g. "Red Hat Enterprise Linux for IBM z Systems"
+
+    :type: str
+    """
+
+    version = attr.ib(type=str, default=None)
+    """Human readable product version string, e.g. "9.4"
+
+    :type: str
+    """
+
+    architecture = attr.ib(type=list, default=None, converter=convert_maybe(sloppylist))
+    """List of architectures supported by the product, e.g. ["s390x"]
+
+    :type: List[str]
+    """
+
+    provided_tags = attr.ib(
+        type=list, default=None, converter=convert_maybe(sloppylist)
+    )
+    """List of tags describing the provided platforms used for pairing with other products,
+    e.g. ["rhel-9", "rhel-9-s390x"]
+
+    :type: List[str]
+    """
+
+
 @attr.s()
 class ProductIdPushItem(PushItem):
     """A :class:`~pushsource.PushItem` representing a product ID certificate.
 
     For push items of this type, the :meth:`~pushsource.PushItem.src` attribute
     refers to a file containing a PEM certificate identifying a product.
+    """
+
+    products = attr.ib(type=list, converter=frozenlist)
+    """List of products described by the ProductID certificate.
```

Comment:
Minor: it would be nice if the doc string included:

```
.. versionadded:: 2.45.0
```

...though we have been rather slack with adding these in general.